catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/PublishedEventHandler.java

```diff
@@ -47,89 +54,110 @@ public PublishedEventHandler(Predicate finalPredicate, Subscription subscription
+            // LOGGER.debug("Catalog framework is in fanout configuration - always evaluate events");
+            // evaluateEvent(event);
+            // }
+            // else if(this.subscription.isEnterprise())
+            if (subscription.isEnterprise()) {
+                // if the subscription is an enterprise subscription then evaluate all incoming events
+                LOGGER.debug("subscription is an enterprise subscription");
                 evaluateEvent(event);
-            } else {
-                LOGGER.debug(
-                        "event received from remote site that is not in list of source IDs of subscription - not evaluating event");
+            } else if (sourceIds == null || sourceIds.isEmpty()) {
+                LOGGER.debug("subscription is a local subscription. Local Source Id: {}",
+                        catalog.getId());
+                if (catalog.getId() != null && catalog.getId()
+                        .equals(metacardSourceId)) {
+                    LOGGER.debug("event received from local site");
+                    evaluateEvent(event);
+                } else {
+                    LOGGER.debug(
+                            "event is from remote site but subscription is local - not evaluating event against subscription filter");
+                }
+            } else if (!sourceIds.isEmpty()) {
+                LOGGER.debug("subscription is a site-based subscription starting with site id {}",
+                        sourceIds.iterator()
+                                .next());
+                // perform site based filtering on subscription
+                if (sourceIds.contains(metacardSourceId)) {
+                    LOGGER.debug("event received from subscribed site");
+                    evaluateEvent(event);
+                } else {
+                    LOGGER.debug(
+                            "event received from remote site that is not in list of source IDs of subscription - not evaluating event");
+                }
             }
-        }
 
-        LOGGER.debug("EXITING: {}", methodName);
-    }
+            LOGGER.debug("EXITING: {}", methodName);
+        }
 
-    private void evaluateEvent(Event event) {
-        // If predicate is NULL then we are handling a filterless subscription - publish all events
-        if (predicate == null) {
-            LOGGER.debug(
-                    "predicate is NULL (must be filterless subscription), publishing all events");
-            new DeliveryProcessor(subscription, preDelivery).process(event);
-            // Otherwise, only send events that match the predicate's filter criteria
-        } else if (predicate.matches(event)) {
-            new DeliveryProcessor(subscription, preDelivery).process(event);
+        private void evaluateEvent(Event event) {
+            // If predicate is NULL then we are handling a filterless subscription - publish all events
+            if (predicate == null) {
```

Comment:
You really only care about the null `predicate` for logging purposes. Could simplify this logic.